### PR TITLE
Add Authentication Token for Frontend Requests

### PR DESCRIPTION
**Pull request by nicolepeverley**
This PR sets up authentication between our frontend and backend with a json web token that is set as a frontend cookie after the user successfully authenticates with Hubspot OAuth. It also fixes the issue we saw in demo 3 where only one user could be logged into the integration and it stayed logged in across incognito tabs and browsers.

https://trello.com/c/28UUSI0O/111-set-up-auth-for-frontend-config-requests

### CODE REVIEW

**Comment by nicolepeverley** [comment]
[Set up auth for frontend config requests](https://trello.com/c/28UUSI0O/111-set-up-auth-for-frontend-config-requests)

File: src/routes/oath.ts [hunk 1]
```diff
@@ -27,19 +28,22 @@ const SAASQUATCH_CLIENT_SECRET = process.env.SAASQUATCH_CLIENT_SECRET;
...
-		return false
-	}
-	else
-	{
-		return tokenStore["userID"] == userID ? true : false;
+// Generate Server Access Token for frontend requests
+const generateAccessToken = (userID: string) => {
+	return jwt.sign({
+		data: userID
+	}, process.env.SERVER_TOKEN_SECRET, { expiresIn: '1h' });
```
Comment (nicolepeverley) [inline src/routes/oath.ts:35]: Do we think 1 hour is sufficient time for the token to have an active login?

File: src/routes/oath.ts [hunk 2]
```diff
@@ -27,19 +28,22 @@ const SAASQUATCH_CLIENT_SECRET = process.env.SAASQUATCH_CLIENT_SECRET;
 
 export const tokenStore: any = {};
```
Comment (nicolepeverley) [inline src/routes/oath.ts:29]: We still need to take out all use of tokenStore but it is used across files and should be done in a follow up ticket.

File: src/index.ts [hunk 3]
```diff
@@ -27,7 +27,8 @@ const PORT = process.env.PORT || 8000;
 const {
 	HUBSPOT_API_KEY: HAPIKEY,
 	SAASQUATCH_API_KEY: SAPIKEY,
-	SAASQUATCH_TENANT_ALIAS: STENANTALIAS
+	SAASQUATCH_TENANT_ALIAS: STENANTALIAS,
+  SERVER_TOKEN_SECRET: SERVER_TOKEN_SECRET,
```
Comment (nicolepeverley) [inline src/index.ts:31]: I added the token secret into my .env file does this need to be adjusted for the production build?

File: src/routes/oath.ts (hunk 1 above)
Comment (tolkamps1) [reply src/routes/oath.ts:35]: Yes

**Review by tolkamps1 — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (MAsimSENG) [reply src/index.ts:31]: Adding it to the Heroku config variables?

**Review by MAsimSENG — APPROVED** [review]
Hi, Nicole great work! Thanks for taking the time to do this over the weekend. 


I just had a quick question in general not a specific line of code:
With this change, whenever we have a front-end request is the global HubspotID variable updated for the current user (or the same if they are the same user)? What happens to the other user? Are they logged out or are we preserving their HusbpotID?  My question arises just due to backend requests that will use the global HubspotID to fetch the access token for the current user before making an API call to Hubspot in response to webhooks.

**Comment by MAsimSENG** [comment]
[Refactor Hubspot OAuth](https://trello.com/c/YWc63K6w/117-refactor-hubspot-oauth)

**Comment by nicolepeverley** [comment]
> Hi, Nicole great work! Thanks for taking the time to do this over the weekend.
> 
> I just had a quick question in general not a specific line of code:
> With this change, whenever we have a front-end request is the global HubspotID variable updated for the current user (or the same if they are the same user)? What happens to the other user? Are they logged out or are we preserving their HusbpotID? My question arises just due to backend requests that will use the global HubspotID to fetch the access token for the current user before making an API call to Hubspot in response to webhooks.

Good question! I haven't changed how the current_user/hubspot ID is getting used in the backend yet. Do we only have one variable for this? Wouldn't that mean that only one user can be logged into the integration at one time? Does the backend webhook calls depend on knowing who is signed in from the local current user, we should probably update this to check the database based on the hubspot ID from the webhook. We can talk more about this today in lab too.

File: src/routes/oath.ts (hunk 2 above)
Comment (nicolepeverley) [reply src/routes/oath.ts:29]: @MAsimSENG will handle this in https://trello.com/c/YWc63K6w/117-refactor-hubspot-oauth

File: src/index.ts (hunk 3 above)
Comment (nicolepeverley) [reply src/index.ts:31]: Thanks added to Heroku 👍

**Pull request merged**